### PR TITLE
Extend animation export modes

### LINKED ISSUE #75: Timeline export with multiple skeletons and modifiers does not seem to work
Attaching project, seems to work fine with Better Collada (Option: Export animations but not multiple actions)
[IK-GRABTEXT.zip](https://github.com/godotengine/godot-blender-exporter/files/2275262/IK-GRABTEXT.zip)

How it should look:
main_timeline

### PR DESCRIPTION
**Pull request by Jason0214**
- [x] default mode, export animation as actions, every node has their own AnimationPlayer
- [x] mode squash actions, export actions in nodes with parent-child relations into one AnimationPlayer
- [x] mode scene animation, export all animation into a AnimationPlayer placed at scene root

### CODE REVIEW

**Comment by Jason0214** [comment]
@sdfgeoff  have  a review

this PR add a new animation export mode as reduz suggested, I named it scene animation, it would export all the animation data into an AnimationPlayer placed at the root of scene tree. 

And I changed the test scenes file structure, make it easy to apply different configuration on test blends.

Another animation mode suggested by reduz in #75 is to make use of the new Animation Playback track,  I think should do it later together with bezier interpolation.  Because, these two feature would be official supported in godot 3.1.  (I think we can make a release for godot 3.0 after cycle material PR and this PR got merged

**Comment by Jason0214** [comment]
by the way,  I made up the 
>mode squash actions, export actions in nodes with parent-child relations into one AnimationPlayer
not sure if it is really needed

File: io_scene_godot/__init__.py [hunk 1]
```diff
@@ -75,43 +75,51 @@ class ExportGodot(bpy.types.Operator, ExportHelper):
     use_export_shape_key = BoolProperty(
         name="Export Shape Key",
         description="Export all the shape keys in mesh objects",
-        default=True,
+        default=False,
```
Comment (sdfgeoff) [inline io_scene_godot/__init__.py]: Why this change?

File: io_scene_godot/__init__.py [hunk 2]
```diff
@@ -75,43 +75,51 @@ class ExportGodot(bpy.types.Operator, ExportHelper):
...
-    use_seperate_animation_player = BoolProperty(
-        name="Seperate AnimationPlayer For Each Object",
-        description="Create a seperate AnimationPlayer node for every"
-                    "blender object which has animtion data",
+    )
+    squash_animation_players = BoolProperty(
+        name="Squash actions",
+        description="Animation is exported as actions of nodes, but instead "
+                    "of having an individual AnimationPlayer for each node, "
+                    "this configuration would squash children nodes' actions "
```
Comment (sdfgeoff) [inline io_scene_godot/__init__.py]: How do these two animation options interact. Would it be better to have an "animation_mode" drop-down that switched between squashed, unsquashed and scene?

Comment (Jason0214) [reply io_scene_godot/__init__.py]: Good suggestion, that should would be much better, I will do it

File: tests/export_test_scenes.py [hunk 3]
```diff
@@ -24,14 +28,22 @@ def main():
             out_path, blend_name = os.path.split(full_path)
             out_path = os.path.join(
                 out_path,
-                '../godot_project/exports/',
+                '../../godot_project/exports/',
```
Comment (sdfgeoff) [inline tests/export_test_scenes.py]: I would expect the folder hierarchy to be preserved between the input and output folder. As far as I read this code (and the lack of changes in the commit), it looks like this still exports everything to the same place.

File: io_scene_godot/__init__.py (hunk 1 above)
Comment (Jason0214) [reply io_scene_godot/__init__.py]: The shape key is usually time consuming to export, so I think it is better that user explicitly check it when they really want to export

File: tests/export_test_scenes.py [hunk 4]
```diff
@@ -24,14 +28,22 @@ def main():
...
                 )
             print(out_path)
-            export_escn(out_path)
+            export_escn(out_path, config)
             print("Exported")
 
 
+def main():
+    test_scenes_dir = os.path.join(os.getcwd(), "tests/test_scenes")
+    for dir_name in os.listdir(test_scenes_dir):
```
Comment (sdfgeoff) [inline tests/export_test_scenes.py]: Only one folder deep. Hmm. Maybe use os.walk, and pass the directory to export_blend (as per above, so preserve the folder structure between the two places).

File: tests/export_test_scenes.py [hunk 5]
```diff
@@ -2,19 +2,23 @@
...
     import io_scene_godot
-    io_scene_godot.export(out_file, {})
+    io_scene_godot.export(out_file, config)
 
 
-def main():
-    target_dir = os.path.join(os.getcwd(), "tests/test_scenes")
+def export_blends(target_dir):
+    config = {}
+    if os.path.exists(os.path.join(target_dir, 'config.json')):
```
Comment (sdfgeoff) [inline tests/export_test_scenes.py]: +1
Nice. Yes, it was needing this.

File: io_scene_godot/__init__.py (hunk 1 above)
Comment (sdfgeoff) [reply io_scene_godot/__init__.py]: But if shapekeys are present in a scene, a user will probably want to export them.

File: io_scene_godot/__init__.py [hunk 6]
```diff
@@ -75,43 +75,51 @@ class ExportGodot(bpy.types.Operator, ExportHelper):
...
+        description="All the animations of the whole scene would be placed "
+                    "into one AnimationPlayer",
         default=False,
     )
     use_mesh_modifiers = BoolProperty(
         name="Apply Modifiers",
         description="Apply modifiers to mesh objects (on a copy!).",
-        default=True,
-        )
+        default=False,
```
Comment (sdfgeoff) [inline io_scene_godot/__init__.py]: Similar as shapekeys. I would leave this enabled by default.

Comment (Jason0214) [reply io_scene_godot/__init__.py]: ok I will change it back

**Comment by Jason0214** [comment]
ready to review again @sdfgeoff , now test_scene can have arbitrary tree structure and would be kept into export directory
